sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/ownership_manager.py

```diff
@@ -157,33 +138,62 @@ def _balance_ownership(self, ownership_list, all_partition_ids):
             active_ownership_by_owner[ownership["owner_id"]].append(ownership)
         active_ownership_self = active_ownership_by_owner[self.owner_id]
 
-        # calculate expected count per owner
+        # calculate expected and max count per owner
         all_partition_count = len(all_partition_ids)
         # owners_count is the number of active owners. If self.owner_id is not yet among the active owners,
         # then plus 1 to include self. This will make owners_count >= 1.
         owners_count = len(active_ownership_by_owner) + (
             0 if self.owner_id in active_ownership_by_owner else 1
         )
         expected_count_per_owner = all_partition_count // owners_count
+        max_count_per_owner = int(math.ceil(all_partition_count / owners_count))
         # end of calculating expected count per owner
 
         to_claim = active_ownership_self
-        if len(active_ownership_self) < expected_count_per_owner:
-            # Either claims an inactive partition, or steals from other owners
-            if claimable_partition_ids:  # claim an inactive partition if there is
-                random_partition_id = random.choice(claimable_partition_ids)
-                random_chosen_to_claim = ownership_dict.get(
-                    random_partition_id,
-                    {
-                        "fully_qualified_namespace": self.fully_qualified_namespace,
-                        "partition_id": random_partition_id,
-                        "eventhub_name": self.eventhub_name,
-                        "consumer_group": self.consumer_group,
-                    },
+        if len(active_ownership_self) < max_count_per_owner:
+            to_try_steal = True
+            if (
+                LoadBalancingStrategy.GREEDY is self.load_balancing_strategy
```

Comment:
This if statement is a bit strange - also why is it in parentheses?
Using `is` is not the same as using `==` - this is checking whether it's the same instance - I don't think that's what we want here?

Reply:
For enum, "is" works the same as ==, for instance:

```python
>>> a = LoadBalancingStrategy('greedy')
>>> b = LoadBalancingStrategy(LoadBalancingStrategy.GREEDY)
>>> a is b
True
>>> LoadBalancingStrategy.GREEDY is a
True
```

Reply:
The parentheses was there before this PR possibly because of a format tool? Anyway I removed it.
Changed to "self.load_balancing_strategy is LoadBalancingStrategy.GREEDY". Some languages prefer placing constants before variables to test equality. It's true that Python doesn't need this. Changed to
```python
if self.load_balancing_strategy is LoadBalancingStrategy.GREEDY:
```